ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Conversions.scala

```diff
@@ -74,7 +74,7 @@ private[state] object Conversions {
       .setCommandDedup(
         DamlCommandDedupKey.newBuilder
           .setCommandId(subInfo.getCommandId)
-          .setSubmitter(subInfo.getSubmitter)
+          .addAllSubmitters(subInfo.getSubmittersList)
```

Comment:
Conceptually, the submitters are a set. For the command deduplication key, the submitters list should have a defined ordering. 

Two options:
- Change the submitters from List to Set on the participant state API (PR https://github.com/digital-asset/daml/pull/8026) and clone+sort the list here. 
- Have the ledger API server sort the list of submitters while validating the command, and document that the submitters are always passed around as an alphabetically sorted list.

Reply:
So far we tend to deal with ordering at the last useful moment (even though sometimes unfortunately we forget) to 1. minimize overhead and 2. make sure the ordering was not inadvertently broken along the way so I'd vote for the second option if changing the ledger API is possible. Else, I'd leave it a list but I'd still perform the ordering here.

Reply:
Can't we change the various APIs to use a `SortedSet` instead?

Reply:
@gerolf-da has been looking into that. Advantage: the ordering invariant is kept automatically. Disadvantages: higher overhead, potentially big change.

Reply:
If non-determinism tests were bulletproof I'd go for the lower overhead option but I am honestly conflicted at this stage, even though for consistency (and to avoid unlikely but potential performance regressions) I'd still stay with our current approach and I'd rather invest a lot in non-determinism testing.

Reply:
After a chat with @gerolf-da and @fabiotudone-da, I have decided to explicitly sort the submitters when building the deduplication key. Almost always, the submitters list will have 1 element, so it should not affect performance.